src/nmea.c

```diff
@@ -78,27 +112,33 @@ void nmea_setup(void)
   SETTING("nmea", "gpgll_msg_rate", gpgll_msg_rate, TYPE_INT);
 }
 
-/** Calculate the checksum of an NMEA sentence.
+/** Calculate and append the checksum of an NMEA sentence.
  * Calculates the bitwise XOR of the characters in a string until the end of
- * the string or an `*` is encountered. If the first character is `$` then it
+ * the string or a `*` is encountered. If the first character is `$` then it
  * is skipped.
  *
- * \param s An NMEA sentence for which to generate the checksum.
- * \return The checksum value.
+ * \param s A null-terminated NMEA sentence, up to and optionally
+ * including the '*'
+ *
+ * \param size Length of the buffer.
+ * 
  */
-u8 nmea_checksum(char *s)
+static void nmea_append_checksum(char *s, size_t size)
 {
   u8 sum = 0;
+  char *p = s;
 
-  if (*s == '$')
-    s++;
+  if (*p == '$')
+    p++;
 
-  while (*s != '*' && *s) {
-    sum ^= *s;
-    s++;
+  while (*p != '*' &&
```

Comment:
Same? `/* '*' between data fields and checksum is not used in checksum calculation. */`